### ISSUE TITLE
Multivariate example models

### ISSUE THREAD

**Issue by ahartikainen** [arviz-devs/arviz#329]
We should have example models that have scalar, vectors, matrices and Nd-matrices (maybe up to 5 dimensions?)

And tests should run against these.

Also values should follow some nice property, like means are in unravelled format as `np.arange(0, N)`?

**Comment by rpgoldman** [comment]
Seconded -- I couldn't figure out how to use arviz when I was doing prior sampling from a regression model, because I ended up with a variable that was a vector of length approximately 1200, which blew everything up.  Not understanding this, I had to revert to turning it into a data frame so I could group and subset it intelligently, and then used pandas to plot.

Another thing that would be nice to have in the examples: when predictor items are in the model as shared numpy vectors, it's not obvious how to get them into the arviz data objects so that one can easily plot them against each other and against the predicted value(s).  I tried crudely adding `pm.Deterministic` variables for the predictors (which were actually used in a rescaled form), and that just caused PyMC3/theano to blow up because they didn't obviously connect to the processing graph.
